src/freenet/client/async/SplitFileInserter.java

```diff
@@ -139,7 +143,9 @@ public void onResume(ClientContext context) throws InsertException, ResumeFailed
                     context.getChkInsertScheduler(realTime).fetchingKeys(), context.persistentFG, 
                     context.persistentFileTracker, context.getPersistentMasterSecret());
             storage.onResume(context);
-            this.sender = new SplitFileInserterSender(this, storage);
+            synchronized(this) {
+                this.sender = new SplitFileInserterSender(this, storage);
```

Comment:
Why does assigning this reference need the lock? (Although I own a copy of _Java Concurrency In Practice_ I have yet to read it.)